### PR TITLE
issue fix https://github.com/Fullscript/hw_admin/issues/8051

### PR DESCRIPTION
**Pull request by glongman**
next: testing in hw_admin staging env - so, another PR

### CODE REVIEW

File: app/models/spree/avalara_transaction.rb [hunk 1]
```diff
@@ -29,7 +29,17 @@ def commit_avatax(invoice_dt = nil, return_auth = nil)
...
+        #   (if gem spree preference is set to true)
+        #   AND
+        #   (
+        #     if the order was never handled by the FullscriptTransactionCalculator)
+        #     OR
+        #     the order tax was successfuly calculated by avata
+        #   )
+        #
+        #  This applies to both orders and returns.
+        if tax_calculation_enabled? && (order.tax_application.blank? || order.tax_application == 'avatax')
```
Comment (cwise) [inline app/models/spree/avalara_transaction.rb:42]: And basically store would be the value on those other ones?

File: spec/models/spree/avalara_transaction_spec.rb [hunk 2]
```diff
@@ -73,27 +74,89 @@
...
+        context 'taxes off globally' do
+          before do
+            allow(order).to receive(:tax_application).and_return 'none'
+          end
+          it_behaves_like 'no commit based on Fullscript order tax application'
         end
+
+        context 'taxes off for store' do
+          before do
+            allow(order).to receive(:tax_application).and_return 'store'
```
Comment (cwise) [inline spec/models/spree/avalara_transaction_spec.rb:139]: 😎

File: app/models/spree/avalara_transaction.rb (hunk 1 above)
Comment (glongman) [reply app/models/spree/avalara_transaction.rb:42]: 'store' and 'none' are the other 2 values that could be set

Comment (kurtfunai) [reply app/models/spree/avalara_transaction.rb:42]: makes sense to me

**Comment by kurtfunai** [comment]
:+1: and nice tests

**Comment by glongman** [comment]
related https://git.fullscript.io/developers/hw-admin/merge_requests/8573

**Comment by glongman** [comment]
note to self.

Before merging....

 In the spree gem, rails 42 changes were merged since the main app was locked on the version (commit) that works for rails 41.

Double check that this was *not* done for this gem and that rails42 changes are in a separate branch and hw_admin branch for rails 42 is using that branch.

**Comment by glongman** [comment]
Tested this branch in a rancher staging.

1. Placed order - taxes calculated and committed ...
2. set `store#collect_taxes` to `false`
3. Place 2nd order - no taxes calculated and order *not* committed.

**Comment by glongman** [comment]
merge check (see previous `note to self` done)
ok to merge

**Pull request merged**